src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -111,53 +187,101 @@ int distance(BoundDecisionDagNode x)
             out bool requiresFalseWhenClause,
             out bool unnamedEnumValue)
         {
+#if DEBUG
+            // Excercise enumeration of all paths to node
+            VisitPathsToNode(nodes, targetNode, nullPaths, handler: (currentPathToNode, currentRequiresFalseWhenClause) => true);
+#endif
+
             unnamedEnumValue = false;
 
             // Compute the path to the node, excluding the node itself.
-            var pathToNode = ShortestPathToNode(nodes, targetNode, nullPaths, out requiresFalseWhenClause);
+            var shortestPathToNode = ShortestPathToNode(nodes, targetNode, nullPaths, out requiresFalseWhenClause);
+            gatherConstraintsAndEvaluations(targetNode, shortestPathToNode, out var constraints, out var evaluations);
+
+            try
+            {
+                return SamplePatternForTemp(rootIdentifier, constraints, evaluations, requireExactType: false, ref unnamedEnumValue);
+            }
+            catch (NoRemainingValuesException)
+            {
+            }
+
+            // In rare cases, the shortest path isn't the one that yields a sample
+            string altSamplePatternForTemp = null;
```

Comment:
>string altSamplePatternForTemp = null; [](http://example.com/codeflow?start=12&length=38)

Consider if we want to extract reminder of the method body into a local function in order to avoid allocating closure for a common (trivial) scenario.